### PR TITLE
PYTHON-2717 Clarify whether maxPoolSize allows 0 option or no

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
We need to test the behavior of `maxPoolSize=0` too. Also please update the spec tests. I assume there are uri-options or CMAP tests for this change?

File: test/test_pooling.py [hunk 1]
```diff
@@ -486,10 +486,6 @@ def f():
         self.assertEqual(nthreads, self.n_passed)
         self.assertTrue(len(cx_pool.sockets) > 1)
 
-    def test_max_pool_size_zero(self):
-        with self.assertRaises(ValueError):
-            rs_or_single_client(maxPoolSize=0)
```
Comment (ShaneHarvey) [inline test/test_pooling.py:491]: We should reinstate this test and assert that `maxPoolSize=0` works as it's expected to.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
I think we have to update the docs for maxPoolSize to reflect that 0 is now a valid value.

File: test/test_pooling.py (hunk 1 above)
Comment (juliusgeo) [reply test/test_pooling.py:491]: Done.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: pymongo/mongo_client.py [hunk 2]
```diff
@@ -207,7 +207,9 @@ def __init__(
           - `maxPoolSize` (optional): The maximum allowable number of
             concurrent connections to each connected server. Requests to a
             server will block if there are `maxPoolSize` outstanding
-            connections to the requested server. Defaults to 100. Cannot be 0.
+            connections to the requested server. Defaults to 100. Can be
+            either 0 or None, in which case there is no limit on the number
+            of concurrent connections.
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply pymongo/mongo_client.py:212]: Done.

File: pymongo/pool.py [hunk 3]
```diff
@@ -1132,7 +1132,7 @@ def __init__(self, address, options, handshake=True):
         self.size_cond = threading.Condition(self.lock)
         self.requests = 0
         self.max_pool_size = self.opts.max_pool_size
-        if self.max_pool_size is None:
+        if self.max_pool_size is None or self.max_pool_size == 0:
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply pymongo/pool.py]: Done.

File: test/test_pooling.py [hunk 4]
```diff
@@ -487,8 +487,35 @@ def f():
...
+                self.n_passed += 1
+
+        for i in range(nthreads):
+            t = threading.Thread(target=f)
+            threads.append(t)
+            t.start()
+
+        joinall(threads)
+        self.assertEqual(nthreads, self.n_passed)
+        self.assertTrue(len(cx_pool.sockets) > 1)
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply test/test_pooling.py]: Done.

**Review by ShaneHarvey — APPROVED** [review]
LGTM once the tests pass. Please also update the title/commit message to reflect the actual change. Something like "PYTHON-2717 Treat maxPoolSize=0 the same as maxPoolSize=None"

**Pull request merged**